File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int max_cost=0,cost=0;
        if(n==0){
            return 0;
        }
        int min_price=prices[0];
        for(int i=0;i<n ;i++){
            min_price= Math.min(min_price,prices[i]);
            cost = prices[i]- min_price;
            max_cost= Math.max(max_cost, cost);
        }
        return max_cost;
    }
}

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        vector<int> dp(cost.size()+3, INT_MAX/1000);
        dp[cost.size()]=0;
        for(int i= cost.size()-1; i>=0;i-- ){
            dp[i]= min(dp[i+1]+cost[i],dp[i+2]+cost[i]);
        }
        return min(dp[0],dp[1]);
    }
};
